tensordict/prototype/tensorclass.py

```diff
@@ -209,18 +215,26 @@ def wrapper(self, *args, batch_size, device=None, **kwargs):
     return wrapper
 
 
-def _from_tensordict_wrapper(expected_keys):
-    def wrapper(cls, tensordict):
+def _to_tensorclass_wrapper(expected_keys):
+    def wrapper(cls, tensordict, non_tensordict=None):
         if not all(key in expected_keys for key in tensordict.keys()):
             raise ValueError(
                 f"Keys from the tensordict ({set(tensordict.keys())}) must "
                 f"correspond to the class attributes ({expected_keys})."
             )
+        if non_tensordict is not None:
+            if not all(key in expected_keys for key in non_tensordict.keys()):
+                raise ValueError(
+                    f"Keys from the non-tensor data ({set(non_tensordict.keys())}) must "
+                    f"correspond to the class attributes ({expected_keys})."
+                )
```

Comment:
I think we miss the check that ensures a key is present either in tensordict or in non_tensordict.
plus we need to check the type.

Reply:
Added the check